docs/docs/user/core-concepts/components/kafka-app.md

```diff
@@ -0,0 +1,41 @@
+# KafkaApp
+
+Subclass of [_KubernetesApp_](kubernetes-app.md).
+
+### Usage
+
+- Defines a [streams-bootstrap](https://github.com/bakdata/streams-bootstrap#usage){target=_blank} component
+- Should not be used in `pipeline.yaml` as the component can be defined as either a [StreamsApp](streams-app.md)
+  or a [ProducerApp](producer-app.md)
+- Often used in `defaults.yaml`
+
+### Configuration
+
+??? example "`pipeline.yaml`"
+
+    ```yaml
+      --8<--
+      ./docs/resources/pipeline-components/kafka-app.yaml
+      --8<--
+    ```
```

Comment:
Dedent to avoid whitespaces in the rendered codeblock

```suggestion
    ```yaml
    --8<--
    ./docs/resources/pipeline-components/kafka-app.yaml
    --8<--
    ```
```

Reply:
@sujuka99 This works fine when rendered and it has the indent on main
https://github.com/bakdata/kpops/blob/main/docs/docs/user/references/components.md

Reply:
I think I am the one who indented it like this in the first place, but it results in useless whitespaces. I'd prefer to remove them if possible.

![image](https://github.com/bakdata/kpops/assets/44573859/5404f96f-797d-4200-8141-419fc2b2a5c9)